src/index.js

```diff
@@ -100,6 +53,13 @@ class IPLDResolver {
         delete this.resolvers[multicodec]
       }
     }
+
+    // Enable all supplied formats
+    for (const format of options.formats) {
+      const {resolver, util} = format
+      const multicodec = resolver.multicodec
+      this.support.add(multicodec, resolver, util)
+    }
```

Comment:
This'll undo the perf work here https://github.com/ipld/js-ipld/pull/145 as IPFS (and potentially others) will have to require these formats upfront.

Can we load these on demand? This would actually be good for IPFS in browsers because we could exclude the implementations from the bundle but add loaders to load in the code when/if needed.

Something like this?:

```js
// In IPFS:
ipld.support.add('git-raw', () => new Promise((resolve, reject) => {
  // Could replace with a XHR request. In Node.js we just:
  resolve(require('ipld-git'))
}))
```
```js
// In ipld
let format
if (typeof this.support[codec] === 'function') {
  format = await this.support[codec]()
  this.support[codec] = format
} else if (this.support[codec]) {
  format = this.support[codec]
} else {
  throw new Error('missing format')
}
```

Reply:
My idea was that you'd normally pass in only the things you need anyway. But then I found out that this is certainly not the case for the js-ifps CLI, where you want support for all formats.

That's a good idea. Thanks for also flushing out the code.